src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerationTemplate.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Routing
+{
+    /// <summary>
+    /// Defines a contract to generate a URL from a template.
+    /// </summary>
+    public abstract class LinkGenerationTemplate
+    {
+        /// <summary>
+        /// Generates a URL with an absolute path.
```

Comment:
Include details about the parameters in the summary when there are overloads. For these two overloads something like:

1. Generates a URL with an absolute path from route values.
2. Generates a URL with an absolute path from route values and the specified link options.

e.g. https://github.com/aspnet/Routing/blob/9f3edca05bb736a15b9e401121779b0e43e388f2/src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternFactory.cs#L579-L582

Do this everywhere in the PR's docs